### ISSUE TITLE
Make the error message for users who may not add datasets/-verses more explicit

### ISSUE THREAD

**Issue by BPeuch** [IQSS/dataverse#6708]
Version: 4.19

Hello everybody,

When a user is not allowed to add datasets or dataverses, clicking on the Add Data button in the upper toolbar leads to a 403 "Not Authorized" page which I feel could be a bit more explicitly worded:

![not_authorized](https://user-images.githubusercontent.com/15209224/75562662-3c5c0f00-5a49-11ea-8608-3d536a5e0395.png)

I think it would be more user-friendly to have something along the lines of: "You do not have permission to add new data[sets/verses]. You can use the Contact button to request permission from the [name of Dataverse] administrator."

@youssefOuahalou explained to me that this is a generic error message and that having a more specific message would require some fine tuning.

However I think it would make things more explicit
- either to remove the `Add Data` button entirely if a user was not granted a role allowing them to create new datasets/-verses — but this would clash with the philosophy outlined in #5718, which I find very productive;

- or to make a custom error message encouraging users to request posting rights.

**Comment by pdurbin** [comment]
@BPeuch I totally agree. Can you please post a screenshot of what the error looks like when you use the **other** (older) "Add Data" button? This one:

![Screen Shot 2020-02-28 at 11 26 21 AM](https://user-images.githubusercontent.com/21006/75566159-3c461a00-5a1d-11ea-8869-b6d8e73bf688.png)

**Comment by BPeuch** [comment]
Well that's the thing, @pdurbin: this other button simply does not appear when you don't have authorization, although the one in the toolbar does. =p

![no_add_data](https://user-images.githubusercontent.com/15209224/75568609-e3de3f00-5a53-11ea-81eb-90e3e9ec1106.png)

I guess the rationale behind hiding this button was: "If they don't have the authorization, then we shouldn't let them believe that they may do it and then confront them with an error message." And then maybe the newer button, in the toolbar, was overlooked.

But either way, I side with your point of view: let users see that it's about adding data, and then encourage them to sign up.

**Comment by pdurbin** [comment]
@BPeuch hmm, is the following issue related?

-  As the creator of a dataverse I'd like to know how to make an "Add Data" button appear for non-logged in users to encourage user interaction #5718

**Comment by LauraHuisintveld** [comment]
For users of DataverseNL, the 'Add Data' button in the navigation bar is very confusing. This button always leads to creating a dataverse or dataset in the root. However, with our set-up, there is not a single user who has the permission to create a dataset or dataverse in the root. (except for a superuser account). 

So the following situation might happen. A user navigates to a subdataverse where he is allowed to add data. He clicks on the "add data" button in the navigation bar and receives an error message, because he is not allowed to add something to the root-dataverse. However, if the same user had clicked on the "add data" button that appears for him below the navigation bar (the older button), he is able to create a dataset. 

Maybe it would be better if the button in the navigation bar would not always lead to adding something in the root, but to add data in the dataverse you are currently viewing on your screen?

**Comment by BPeuch** [comment]
It is related, @pdurbin, in that I agree with your idea of "encourag[ing] user engagement".

@LauraHuisintveld: Very much agreed. Thanks for bringing up this issue.

**Comment by poikilotherm** [comment]
We discussed this just this morning as we had some users confused when trying to add data to the root dataverse of Jülich DATA (which is not allowed).

We were thinking maybe instead of very generic error message, the page could generate a list of dataverses the user actually has permission to add things as links, so the UX is not "getting stuck and doesn't work (TM)".

**Comment by djbrooke** [comment]
Thanks all for the feedback. We're planning on working on the workflows for adding data in the near future, and this was a first step. It looks like we need to make some adjustments here. I'll check in with @TaniaSchlatter and other members of the design team for their thoughts about a short term display logic solution. 

@poikilotherm I don't think we want to display all the dataverses that the user can add data as it could be a very long message. Please hold off on working on this until we get some guidance from the design team. Thanks!

**Comment by poikilotherm** [comment]
@djbrooke it could be easily truncated to let's say the top 10 or however. Pretty sure there are multitude of ways how to do this for a good UX. Confident @TaniaSchlatter et al will come up with sth. brilliant for all of this. :smile:

**Comment by BPeuch** [comment]
@djbrooke wrote:
> I don't think we want to display all the dataverses that the user can add data as it could be a very long message.

When I read that I thought "How bad can it be?" Then I thought I'd check a mature Dataverse, like Harvard's, and indeed… :sweat_smile:

![amount_dataverses](https://user-images.githubusercontent.com/15209224/75693507-4ed38400-5ca7-11ea-9b08-d7a4aa7d09f6.png)

**Comment by kaitlinnewson** [comment]
For Scholars Portal we unfortunately have to remove this 'Add Data' button before we upgrade to 4.19 as we never allow users to post in the root (similar to @LauraHuisintveld). Shouldn't this only appear if they have permissions to publish in root, or be applied to the specific Dataverse that the user is currently in? For our context, we want users to publish in their specific dataverse for their institution, but of course this varies by installation.

**Comment by djbrooke** [comment]
I'm going to close this as a duplicate of #6741, where we've started to evaluate how we can change display logic to support these cases.

**Comment by TaniaSchlatter** [comment]
Also related to the solution described in #6772
